### PR TITLE
Balder/reduce timestamp usage

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

File: vespalib/src/vespa/vespalib/util/time.cpp [hunk 1]
```diff
@@ -11,6 +12,42 @@ to_utc(steady_time ts) {
...
+}
+
+string
+to_string(system_time time) {
+    return to_string(time.time_since_epoch());
+}
+
+string
+to_string(steady_time time) {
+    return to_string(time.time_since_epoch());
```
Comment (vekterli) [inline vespalib/src/vespa/vespalib/util/time.cpp]: Additionally, a test of this functionality would be nice

**Review by vekterli — COMMENTED** [review]
Overall looks good, but see comment regarding string conversion of time points/durations.

File: searchsummary/src/tests/extractkeywords/extractkeywordstest.cpp [hunk 2]
```diff
@@ -109,7 +109,7 @@ ExtractKeywordsTest::Main()
             }
 
     // Print time taken
-    double timeTaken = timer.elapsed().ms();
+    double timeTaken = vespalib::count_ms(timer.elapsed());
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchsummary/src/tests/extractkeywords/extractkeywordstest.cpp]: Fixed

File: searchlib/src/tests/fef/phrasesplitter/benchmark.cpp [hunk 3]
```diff
@@ -74,8 +72,8 @@ Benchmark::Main()
 
     run(numRuns, numPositions);
 
-    std::cout << "TET:  " << _sample.ms() << " (ms)" << std::endl;
-    std::cout << "ETPD: " << std::fixed << std::setprecision(10) << _sample.ms() / numRuns << " (ms)" << std::endl;
+    std::cout << "TET:  " << vespalib::count_ms(_sample) << " (ms)" << std::endl;
+    std::cout << "ETPD: " << std::fixed << std::setprecision(10) << vespalib::count_ms(_sample) / numRuns << " (ms)" << std::endl;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/tests/fef/phrasesplitter/benchmark.cpp]: Fixed

File: searchcore/src/vespa/searchcore/proton/server/storeonlydocsubdb.cpp [hunk 4]
```diff
@@ -394,14 +393,14 @@ StoreOnlyDocSubDB::updateLidReuseDelayer(const DocumentDBConfig * newConfigSnaps
 void
 StoreOnlyDocSubDB::updateLidReuseDelayer(const LidReuseDelayerConfig &config)
 {
-    bool immediateCommit = config.visibilityDelay() == 0;
+    bool immediateCommit = config.visibilityDelay() == vespalib::duration::zero();
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/storeonlydocsubdb.cpp]: Fixed

File: searchcommon/src/vespa/searchcommon/common/schema.cpp [hunk 5]
```diff
@@ -488,71 +476,39 @@ struct IntersectHelper {
...
-bool IntersectHelper::is_matching(const Schema::FieldSet &f1,
-                                  const Schema::FieldSet &f2) {
+bool IntersectHelper::is_matching(const Schema::FieldSet &f1, const Schema::FieldSet &f2) {
     if (f1.getFields() != f2.getFields())
         return false;
-    const std::vector<vespalib::string> fields = f1.getFields();
-    for (std::vector<vespalib::string>::const_iterator
-             i = fields.begin(), ie = fields.end(); i != ie; ++i) {
-        if (schema->getIndexFieldId(*i) == Schema::UNKNOWN_FIELD_ID) {
+    for (const vespalib::string & field : f1.getFields()) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcommon/src/vespa/searchcommon/common/schema.cpp]: :)

File: metrics/src/vespa/metrics/state_api_adapter.cpp [hunk 6]
```diff
@@ -30,7 +29,7 @@ StateApiAdapter::getTotalMetrics(const vespalib::string &consumer)
     _manager.updateMetrics(true);
     metrics::MetricLockGuard guard(_manager.getMetricLock());
     _manager.checkMetricsAltered(guard);
-    time_t currentTime = fastos::time();
+    time_t currentTime = vespalib::to_s(vespalib::steady_clock::now().time_since_epoch());
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply metrics/src/vespa/metrics/state_api_adapter.cpp]: Fixed

File: metrics/src/vespa/metrics/metricmanager.cpp [hunk 7]
```diff
@@ -26,7 +26,7 @@ MetricManager::ConsumerSpec::~ConsumerSpec() = default;
 
 time_t
 MetricManager::Timer::getTime() const {
-    return fastos::time();
+    return vespalib::to_s(vespalib::steady_clock::now().time_since_epoch());
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply metrics/src/vespa/metrics/metricmanager.cpp]: Fixed

File: logd/src/logd/watcher.cpp [hunk 8]
```diff
@@ -266,13 +266,14 @@ Watcher::watchfile()
...
-            LOG(debug, "preparing to rotate logfile, old logfile size %d, age %d seconds",
-                (int)offset, (int)(now-created));
+            rotTimer = vespalib::Timer();
+            LOG(debug, "preparing to rotate logfile, old logfile size %d, age %2.3f seconds",
+                (int)offset, vespalib::to_s(now-created));
             char newfn[FILENAME_MAX];
             int l = strlen(filename);
             strcpy(newfn, filename);
-            struct tm *nowtm = gmtime(&now);
+            time_t seconds = vespalib::to_s(now.time_since_epoch());
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply logd/src/logd/watcher.cpp]: Fixed

File: vespalib/src/vespa/vespalib/util/time.cpp [hunk 9]
```diff
@@ -11,6 +12,42 @@ to_utc(steady_time ts) {
...
+}
+
 Timer::~Timer() = default;
 
+void
+Timer::waitAtLeast(duration dur, bool busyWait) {
+    if (busyWait) {
+        steady_clock::time_point deadline = steady_clock::now() + dur;
+        while (steady_clock::now() < deadline) {
+            for (int i = 0; i < 1000; i++) { }
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply vespalib/src/vespa/vespalib/util/time.cpp]: Fixed

File: vespalib/src/vespa/vespalib/util/time.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (baldersheim) [reply vespalib/src/vespa/vespalib/util/time.cpp]: Correct, and correct. I have fixed this on the call site where I need to cheat (for testing).

Comment (baldersheim) [reply vespalib/src/vespa/vespalib/util/time.cpp]: Added.

**Pull request merged**